encoders/firebase-encoders-json/src/json/java/com/google/firebase/encoders/json/JsonValueObjectEncoderContext.java

```diff
@@ -37,36 +37,37 @@
   private final Map<Class<?>, ObjectEncoder<?>> objectEncoders;
   private final Map<Class<?>, ValueEncoder<?>> valueEncoders;
   private final ObjectEncoder<Object> fallbackEncoder;
+  private final boolean ignoreNullValues;
 
   JsonValueObjectEncoderContext(
       @NonNull Writer writer,
       @NonNull Map<Class<?>, ObjectEncoder<?>> objectEncoders,
       @NonNull Map<Class<?>, ValueEncoder<?>> valueEncoders,
-      ObjectEncoder<Object> fallbackEncoder) {
+      ObjectEncoder<Object> fallbackEncoder,
+      boolean ignoreNullValues) {
     this.jsonWriter = new JsonWriter(writer);
     this.objectEncoders = objectEncoders;
     this.valueEncoders = valueEncoders;
     this.fallbackEncoder = fallbackEncoder;
+    this.ignoreNullValues = ignoreNullValues;
   }
 
   private JsonValueObjectEncoderContext(JsonValueObjectEncoderContext anotherContext) {
     this.jsonWriter = anotherContext.jsonWriter;
     this.objectEncoders = anotherContext.objectEncoders;
     this.valueEncoders = anotherContext.valueEncoders;
     this.fallbackEncoder = anotherContext.fallbackEncoder;
+    this.ignoreNullValues = anotherContext.ignoreNullValues;
   }
 
   @NonNull
   @Override
   public JsonValueObjectEncoderContext add(@NonNull String name, @Nullable Object o)
       throws IOException, EncodingException {
-    maybeUnNest();
-    jsonWriter.name(name);
-    if (o == null) {
-      jsonWriter.nullValue();
-      return this;
+    if (ignoreNullValues) {
```

Comment:
I think you can go further simplify by just adding here

```
if (ignoreNullValues && o == null) return this;
```

and the rest of the code can stay the same.

Reply:
The reason I didn't go with that approach is that, in the case `ignoreNullValues` is `true`, and `o` is _not_ `null`, it will be `null`-checked a second time unnecessarily.

Reply:
Actually, both methods `internalAdd` and `internalAddIgnoreNullValues` have a `if (o == null)` condition, so the second check still happens.

Reply:
Yes, in either case, `o` gets `null`-checked _once_. But if I just added that line to the original method, `o` would get `null`-checked _again_ after `jsonWriter.name(name);`

Reply:
TL;DR; LGTM

Long form: Actually, the conditional will short-circuit if `ignoreNullValues==false`, so the "first check" will *only* happen when `ignoreNullValues=true` and `o != null`, which most like is going to be **most common case** :-)

Reply:
I think this is essentially what I was saying before. :)

> The reason I didn't go with that approach is that, in the case `ignoreNullValues` is `true`, and `o` is _not_ `null`, it will be `null`-checked a second time unnecessarily.